medical-system/src/components/shared/Table.tsx

```diff
@@ -0,0 +1,128 @@
+import { ReactNode, useEffect, useState } from "react";
+import {
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import { collection, getDocs } from "firebase/firestore";
+import { db } from "../../auth/firebase-config";
+
+export type User = {
+  name: string;
+  email: string;
+  mobile: string;
+  address: string;
+  dateBirth: string;
+  delete?: ReactNode;
+  edit?: ReactNode;
+};
+
+const columns = [
+  {
+    header: "Name",
+    accessorKey: "name",
+  },
+  {
+    header: "Email",
+    accessorKey: "email",
+  },
+  {
+    header: "Mobile",
+    accessorKey: "mobile",
+  },
+  {
+    header: "Address",
+    accessorKey: "address",
+  },
+  {
+    header: "Date Birth",
+    accessorKey: "dateBirth",
+  },
+];
+
+export function Table({ addPatient }: { addPatient: User | null }) {
+  const [data, setData] = useState<User[]>([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const patientsCollectionRef = collection(db, "patients");
+      const users = await getDocs(patientsCollectionRef);
+
+      const usersList = users.docs.map((doc) => {
+        const docData = doc.data();
+        console.log(docData);
+        return {
+          name: docData.name || "",
+          email: docData.email,
+          mobile: docData.mobile || "",
+          address: docData.address || "",
+          dateBirth: docData.dateBirth,
+        } as User;
+      });
+
+      setData(usersList);
+    };
+
+    fetchData();
+  }, []);
```

Comment:
A może pomyślelibyśmy tutaj o react-query? lub przynajmniej useSWR, załatwiłoby to nam isLoading, handling errorow i nie musielibysmy robić tego w useEffect a jedynie jako hook ;)

useSWR zdecydowanie by wystarczyło